experimental/LieAlgebras/src/RootSystem.jl

```diff
@@ -1462,6 +1462,69 @@ function tensor_product_decomposition(
   )
 end
 
+###############################################################################
+#demazures charcter formula
+function _demazure_operator(r::RootSpaceElem, w::WeightLatticeElem)
+  fl, index_of_r = is_simple_root_with_index(r)
+  @req fl "not a simple root"
+  
+  d = 2*dot(w, r)//dot(r, r)
+  list_of_occuring_weights = WeightLatticeElem[]
+  
+  refl = reflect(w, index_of_r)
+
+  
+  if d > -1
+    while w != refl
+      push!(list_of_occuring_weights, w)
+      w -= WeightLatticeElem(r)
+    end
+    push!(list_of_occuring_weights, w)
+    return 1, list_of_occuring_weights
+  elseif d < -1
+    w += WeightLatticeElem(r)
+    push!(list_of_occuring_weights, w)
+    while w != refl-WeightLatticeElem(r)
+      w +=  WeightLatticeElem(r)
+      push!(list_of_occuring_weights, w)
+    end
+    return -1, list_of_occuring_weights
+  else
+    return 0, list_of_occuring_weights
+  end
+end 
+
+function demazure_operator(r::RootSpaceElem, w::WeightLatticeElem)
+  return demazure_operator(r, Dict(w => 1))
+end
+
+function demazure_operator(r::RootSpaceElem, groupringelem::Dict{WeightLatticeElem, Int})
+  dict = Dict{WeightLatticeElem,Int}()
+  for (w, dim) in groupringelem
+    sign, weights = _demazure_operator(r, w)
+    for w in weights
+      val = get(dict, w, 0) + sign * dim
+      if !is_zero(val)
+        dict[w] = val
+      end
```

Comment:
If `val` is zero, you still need to delete the index `w` from `dict`.
```suggestion
      if is_zero(val)
        delete!(dict, w)
      else
        dict[w] = val
      end
```